activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -475,6 +475,12 @@ def replace_on_target(record, skip_callbacks, replace:, inversing: false)
           @_was_loaded = nil
         end
 
+        # This check prevents the accidental creation of duplicate child records
+        # when inverse_of is in use for an already existing owner.
+        def new_record_inverse_of_existing_owner?(record, inversing: false)
```

Comment:
The name is a bit of a mouthful, how about `inverse_of_existing_owner`

```suggestion
        def inverse_of_existing_owner?(record, inversing: false)
```

Also, can you move this under the private namespace, we don't want applications calling this directly.

Reply:
I renamed the method and moved it to the private namespace.